cmd/fs-v1.go

```diff
@@ -113,12 +113,48 @@ func fsReaddirMetaBuckets(fsPath string) ([]string, error) {
 	return f.Readdirnames(-1)
 }
 
+// List of all bucket metadata configs.
 var bucketMetadataConfigs = []string{
 	bucketNotificationConfig,
 	bucketListenerConfig,
 	bucketPolicyConfig,
 }
 
+func migrateBucketMetadataConfigs(metaBucket, bucket, tmpBucket string) error {
+	for _, bucketMetaFile := range bucketMetadataConfigs {
+		_, err := fsStatDir(pathJoin(metaBucket, tmpBucket, bucketMetaFile))
```

Comment:
No we shouldn't os.Stat() is not the right thing to do here. We are making sure that this code works across windows as well. `fsStat()` is nothing but `os.Stat()` but with minio only layered requirements. 

I don't see how os.Stat() would provide finer control.